[Yuanxi] iP

yuanxi1 opened the pull request:
# Duke Task App
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😸 

And it's **FREE**

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
```
}

Commit:
Add Gradle support

Commit:
Implement skeletal version

Commit:
Add store and display tasks functionality

Commit:
Allow marking task as done or undone

Commit:
Add todo, event, deadline task types

Commit:
Fix extra whitespace issue

Commit:
Add simple tests

Commit:
Refactor & Add exception handling

Commit:
Add delete functionality

Commit:
Clean up file structure

Commit:
Add Storage class skeleton and TaskList class

Commit:
Add data folder to .gitignore

Commit:
Add functions to save & load tasks from disk

Commit:
Merge branch 'branch-Level-7'

Commit:
Change to LocalDate from String for dates

Commit:
Resolve merge conflicts

Commit:
Refactor the code to make it more OOP

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add graddle support

Commit:
Remove extra empty line

Commit:
Remove next line character in error messages

Commit:
Modify DukeException constructor to use the parent constructor

Commit:
Delete unnecessary code

Commit:
Add Junit tests for two methods

Commit:
Add javadoc for half of the classes

Commit:
Modify naming to follow coding-standard

Commit:
Modify indentation level to follow coding-standard

Commit:
Add ability to find tasks based on description

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Simplify list task function to use toString method in TaskList class

Elben85 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+import java.util.Scanner;
+
+public class Duke {
+    public static TaskList taskList;
+    public Ui ui = new Ui();
+    public Parser parser = new Parser();
```
Might it be better to remove the static modifier from taskList? (Since a different instance should have a different tasklist)

Also,
I feel the access modifier could be changed to private?

Elben85 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+        }
+    }
+
+    public void chat (Scanner sc) {
```
I think there should't be any whitespace between function names and the function params. I noticed this issue in several other places as well

paullee18 replied:
Might want to remove the whitespace between chat and the arguments
```suggestion
    public void chat(Scanner sc) {
```

Elben85 commented on src/main/java/duke/handlers/DukeCommandMap.java:
```diff
@@ -0,0 +1,26 @@
+        commandMap.put("event", new AddEventCommand());
+        commandMap.put("delete", new DeleteTaskCommand());
+        commandMap.put("find", new FindTaskCommand());
+    }
```
I like how you use hashmap to map the commands since it make the codes looks more simple

Elben85 commented on src/main/java/duke/models/DeadlineParser.java:
```diff
@@ -0,0 +1,41 @@
+    private final String DEADLINE_FORMAT_ERROR =
+            "Deadline must be in this format: <Description> /by <DateTime>";
+    private final String DATE_FORMAT_ERROR =
+            "Your date must be a valid date in dd/MM/yyyy format";
```
Could this be changed to static? I noticed this on other places as well

Elben85 commented on src/main/java/duke/models/DeadlineParser.java:
```diff
@@ -0,0 +1,41 @@
+        if (!pattern.matcher(content).find()) {
+            throw new DukeException(DEADLINE_FORMAT_ERROR);
+        }
+        String[] detail = content.split("\\s/by\\s", 2);
```
Since this represents a collection, shouldn't this be plural?

Elben85 commented on src/main/java/duke/models/EventParser.java:
```diff
@@ -0,0 +1,42 @@
+     * @return The event object represented by the user input string.
+     * @throws DukeException If the user input does not follow the required patterns.
+     */
+    public Event parseEvnet(String content) throws DukeException {
```
Might this be a typo?

Elben85 commented on src/main/java/duke/models/TaskList.java:
```diff
@@ -0,0 +1,66 @@
+        return taskList.remove(taskIndex);
+    }
+
+    public int size() {
```
If I remember correctly, method names should be verb?

Elben85 commented on src/main/java/duke/models/Todo.java:
```diff
@@ -0,0 +1,46 @@
+     * @param isDone The status of the deadline task.
+     */
+    public Todo(String description, boolean isDone) {
+        super(description,isDone);
```
I think there should be whitespace between function arguments?

Elben85 commented on src/main/java/duke/models/Ui.java:
```diff
@@ -0,0 +1,53 @@
+     */
+    public void greet() {
+        System.out.println("Hello from\n" + logo);
+        System.out.print("Tell me what you need\n");
```
Perhaps you can change this line to
`System.out.println("Tell me what you need")`

Elben85 reviewed (commented):
Overall, I think the code structure (the OOP) is already great, but there are definitely some styling improvements that could be done.

paullee18 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+import java.util.Scanner;
+
+public class Duke {
+    public static TaskList taskList;
```
Perhaps taskList should not be a static variable?
```suggestion
    public TaskList taskList;
```

paullee18 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+import duke.models.Ui;
+
+import java.util.Scanner;
+
```
I think adding Javadocs would be good

paullee18 commented on src/main/java/duke/models/Storage.java:
```diff
@@ -0,0 +1,101 @@
+ * @author Zhu Yuanxi
+ */
+public class Storage {
+    private static final String TASK_DIRECTORY_NAME = "src/main/data/";
```
Perhaps file address should be passed into the constructor so that Storage can be used for other files as well

paullee18 reviewed (commented):
Overall, I found your code good except for some minor coding standard violations.

Commit:
Update build.gradle

Commit:
Add simple GUI

Commit:
Add assertion

Commit:
Delete unused import statements

Commit:
Use name constant

yuanxi1 committed:
Merge pull request #3 from yuanxi1/branch-A-Assertion

Add assertion

yuanxi1 committed:
Merge pull request #2 from yuanxi1/branch-A-CodeQuality

Improve code quality

Commit:
Add functionality to create recurring event

User can create recurring event by specifying startDate and endDate with a default weekly interval

Commit:
Add better GUI

Commit:
Refactor response into DukeResponse

Commit:
Add more java doc

Commit:
Fix spelling

Commit:
Update README.md

Commit:
Add Ui image

Commit:
Make Duke main method the entry point for gui

Commit:
Delete unused import

Commit:
Add help command

Commit:
Reorganise file structure

Commit:
Update README.md

Commit:
Update help command response message
